src/scores/continuous/standard_impl.py

```diff
@@ -156,6 +206,35 @@ def mae(
         Alternatively, an xarray structure with dimensions preserved as appropriate
         containing the score along reduced dimensions
     """
+    fcst_is_pandas = isinstance(fcst, pandas.Series)
+    obs_is_pandas = isinstance(obs, pandas.Series)
+    both_pandas = fcst_is_pandas and obs_is_pandas
+    either_pandas = fcst_is_pandas or obs_is_pandas
+    fcst_is_xarray = isinstance(fcst, (xarray.Dataset,xarray.DataArray))
+    obs_is_xarray = isinstance(obs, (xarray.Dataset,xarray.DataArray))
+    either_xarray = fcst_is_xarray or obs_is_xarray
+    mixed_pandas_xarray = either_pandas and either_xarray
+
+    if isinstance(fcst, pandas.Series):
+
+        fcst = fcst.to_xarray()        
+        if mixed_pandas_xarray:
+            series_name = fcst.name        
+            if series_name is None:
+                raise ValueError(PANDAS_NAME_MATCHING_WARNING)
+            
+            fcst = fcst.rename({'index': series_name})
+
+    if isinstance(obs, pandas.Series):
+
+        obs = obs.to_xarray()
+
+        if mixed_pandas_xarray:
+            series_name = fcst.name
+            if series_name is None:
+                raise ValueError(PANDAS_NAME_MATCHING_WARNING)
+        
+            obs = obs.rename({'index': series_name})
```

Comment:
I agree with @benowen-bom, at the very least this behaviour should be in its own function.